storage/src/nodestore.rs

```diff
@@ -1727,23 +1741,33 @@ impl<T, S: ReadableStorage> NodeStore<T, S> {
         start_area_index: AreaIndex,
     ) -> impl Iterator<Item = Result<(LinearAddress, AreaIndex), FileIoError>> {
         self.free_list_iter_inner(start_area_index)
-            .map(|item| item.map(|(addr, area_index, _)| (addr, area_index)))
+            .map(|item| item.map(|(addr, area_index, _, _)| (addr, area_index)))
     }
 
     // pub(crate) since checker will use this to verify that the free areas are in the correct free list
     // Since this is a low-level iterator, we avoid safe conversion to AreaIndex for performance
     pub(crate) fn free_list_iter_inner(
         &self,
         start_area_index: AreaIndex,
-    ) -> impl Iterator<Item = Result<(LinearAddress, AreaIndex, AreaIndex), FileIoError>> {
+    ) -> impl Iterator<
+        Item = Result<(LinearAddress, AreaIndex, AreaIndex, FreeListParentPtr), FileIoError>,
```

Comment:
This is way too big for a tuple, it should be named. Do we really need all these return values?